In [1]:
k = .04, 5, 140
u0 = -20
v0 = -70
vthr = 30
a, b, c, d = .02, -.1, -55, 6

$$
f\left( v \right) = k_1 \cdot v^2 + k_2 \cdot v + k_3
$$

In [2]:
def f(v):
    return k[0] * (v**2) + k[1] * v + k[2]

$$
\dot{v} = f\left( v \right) - u + I
$$

In [3]:
def Vt(v, u, I):
    return f(v) - u + I

$$
\dot{u} = a \cdot \left( b \cdot v - u \right)
$$

In [4]:
def Ut(v, u):
    return a * (b * v - u)

$$
v \approx v_{thr}
\Longrightarrow
\begin{cases}
v \rightarrow c \\
u \rightarrow u + d
\end{cases}
$$

# Isoclines

$$
\begin{cases}
0 = \dot{v} = f\left( v \right) - u + I \\
0 = \dot{u} = a \cdot \left( b \cdot v - u \right)
\end{cases}
\Longrightarrow
\begin{cases}
k_1 \cdot v^2 + k_2 \cdot v + k_3 - u + I = 0\\
a \cdot \left( b \cdot v - u \right) = 0
\end{cases}
\Longrightarrow
\begin{cases}
k_1 \cdot v^2 + v \cdot \left( k_2 - b \right) + k_3 + I = 0\\
b \cdot v = u
\end{cases}
$$

Discriminant for the first equation
$$
D = \left( k_2 - b \right)^2 - 4 \cdot k_1 \cdot \left( k_3 + I \right)
$$

Thus
$$
\begin{cases}
v_{1, 2} = \frac{-b \pm \sqrt{\left( k_2 - b \right)^2 - 4 \cdot k_1 \cdot \left( k_3 + I \right)}}{2 \cdot k_1} \\
u_{1, 2} = v_{1, 2} \cdot b
\end{cases}
$$

Critical value for $I$ can be fetched from the equation
$$
\left( k_2 - b \right)^2 - 4 \cdot k_1 \cdot \left( k_3 + I \right) = 0
$$
Result is
$$
I_{cr} = \frac{\left( k_2 - b \right)^2}{4 \cdot k_1} - k_3
$$

In [5]:
I_critical = ((k[1] - b)**2) / (4 * k[0]) - k[2]; I_critical

22.56249999999997

In [6]:
def v_critical(I):
    return (- b + (b**2 - 4 * k[0] * (k[1] + k[2] + I))) / (2 * k[0])

In [7]:
def u_critical(I):
    v = v_critical(I)
    return v[0] * b, v[1] * b

# Critical values

$$
\begin{cases}
v_{cr} = \frac{-b}{2 \cdot k_1} \\
u_{cr} = \frac{-b^2}{2 \cdot k_1} = b \cdot v_{cr}
\end{cases}
$$

# Linearization

$$
\begin{cases}
\dot{v} = k_1 \cdot v^2 + k_2 + k_3 - u + I \\
\dot{u} = a \cdot \left( b \cdot v - u \right)
\end{cases}
$$

Jacobian
$$
J =
\begin{bmatrix}
k_1 \cdot v & -1 \\
a \cdot b & -a
\end{bmatrix}
$$

Linearized Jacobian
$$
J_l =
\begin{bmatrix}
\frac{-b}{2} & -1 \\
a \cdot b & -a
\end{bmatrix}
$$

Characteristics polynomial
$$
\begin{vmatrix}
\frac{-b}{2} - \lambda & -1 \\
a \cdot b & -a - \lambda
\end{vmatrix}
= \left( \frac{b}{2} + \lambda \right) \cdot \left( a + \lambda \right) + a
= \frac{a \cdot b}{2} + \frac{b \cdot \lambda}{2} + \lambda \cdot a + \lambda^2 + a
= \lambda^2 + \lambda \cdot \frac{b + 2 \cdot a}{2} + a \cdot \frac{b + 2}{2}
$$

Discriminant
$$
D = \left( \frac{b + 2 \cdot a}{2} \right)^2 - 2 \cdot a \cdot \left( b + 2 \right)
$$
Solutions
$$
\lambda_{1, 2} = \frac{- \frac{b + 2 \cdot a}{2} \pm \sqrt{\left( \frac{b + 2 \cdot a}{2} \right)^2 - 2 \cdot a \cdot \left( b + 2 \right)}}{2}
$$

In [8]:
first = (b + 2*a) / 2
second = first**2 - 2 * a * (b + 2)
(- first + second**.5)/2, (- first - second**.5)/2

((0.01500000000000001+0.1370218960604472j),
 (0.014999999999999993-0.1370218960604472j))

$\lambda_{1, 2} = 0.015 \pm 0.137i$ &mdash; eigenvalues are complex-conjugated, real part is positive &mdash; we have unstable focus [equilibrium](http://www.scholarpedia.org/article/Equilibrium)

# Eigenvectors

By definition eigenvectors $e_{1, 2}$ can be found from the equation
$$
\begin{bmatrix}
\frac{-b}{2} - \lambda & -1 \\
a \cdot b & -a - \lambda
\end{bmatrix}
\cdot v
= v \cdot \lambda
$$
In scalar form
$$
\begin{bmatrix}
e_x \cdot \left( \frac{-b}{2} - \lambda \right) - e_y \\
e_x \cdot a \cdot b - e_y \cdot \left( a + \lambda \right)
\end{bmatrix}
= \begin{bmatrix} e_x \\ e_y \end{bmatrix}
\cdot \lambda
$$
Move right part to left
$$
\begin{bmatrix}
e_x \cdot \left( \frac{-b}{2} - 2 \cdot \lambda \right) - e_y \\
e_x \cdot a \cdot b - e_y \cdot \left( a + 2 \cdot \lambda \right)
\end{bmatrix}
= \begin{bmatrix} 0 \\ 0 \end{bmatrix}
$$
Let's rewrite this as a product
$$
\begin{bmatrix}
\left( \frac{-b}{2} - 2 \cdot \lambda \right) & -1 \\
a \cdot b & - \left( a + 2 \cdot \lambda \right)
\end{bmatrix}
\begin{bmatrix} e_x \\ e_y \end{bmatrix}
= \begin{bmatrix} 0 \\ 0 \end{bmatrix}
$$
We can easily find $e_y$
$$
e_y = e_x \cdot \left( \frac{-b}{2} - 2 \cdot \lambda \right)
$$
As the rank of matrix is $1$ because determinant is $0$ by condition of eigenvalues,
we can have eigenvectors with $e_x = -2$
$$
e_{1, 2}
= \cdot \begin{bmatrix}
- 2 \\
b + 4 \cdot \lambda_{1, 2}
\end{bmatrix}
= \begin{bmatrix}
- 2 \\
b - b - 2 \cdot a \pm 2 \cdot \sqrt{\left( \frac{b + 2 \cdot a}{2} \right)^2 - 2 \cdot a \cdot \left( b + 2 \right)}
\end{bmatrix}
= \begin{bmatrix}
- 2 \\
- a \pm \sqrt{\left( \frac{b + 2 \cdot a}{2} \right)^2 - 2 \cdot a \cdot \left( b + 2 \right)}
\end{bmatrix}
$$